preloader/preload.go

```diff
@@ -143,19 +152,36 @@ func writeDaisyWorkflow(inputWorkflow string, outputImage *config.Image, buildSp
 		return "", err
 	}
 	acceleratorsJSON, err := json.Marshal([]map[string]interface{}{})
+	if err != nil {
+		return "", err
+	}
 	if buildSpec.GPUType != "" {
 		acceleratorType := fmt.Sprintf("projects/%s/zones/%s/acceleratorTypes/%s",
 			buildSpec.Project, buildSpec.Zone, buildSpec.GPUType)
 		acceleratorsJSON, err = json.Marshal([]map[string]interface{}{
 			{"acceleratorType": acceleratorType, "acceleratorCount": 1}})
-	}
-	if err != nil {
-		return "", err
+		if err != nil {
+			return "", err
+		}
 	}
 	licensesJSON, err := json.Marshal(outputImage.Licenses)
 	if err != nil {
 		return "", err
 	}
+
+	// template content for the step resize-disk. If disk-size is not provided
+	// a place holder is used, because ResizeDisk API requires a larger size than
+	// the original disk.
+	var resizeDiskJSON string
+	if buildSpec.DiskSize != 0 {
+		// actual disk size
+		resizeDiskJSON = fmt.Sprintf("\"ResizeDisks\": [{\"Name\": \"boot-disk\","+
+			"\"SizeGb\": \"%d\"}]", buildSpec.DiskSize)
```

Comment:
You can use Go raw strings (backticks) to avoid escaping the quotes:

`"ResizeDisks": [{"Name": "boot-disk", "SizeGb": "%d"}]`